include/graaflib/algorithm/shortest_path.tpp

```diff
@@ -97,8 +97,15 @@ std::optional<graph_path<WEIGHT_T>> dijkstra_shortest_path(
     }
 
     for (const auto& neighbor : graph.get_neighbors(current.id)) {
-      WEIGHT_T distance = current.dist_from_start +
-                          get_weight(graph.get_edge(current.id, neighbor));
+      WEIGHT_T edge_weight = get_weight(graph.get_edge(current.id, neighbor));
+
+      if (edge_weight < 0) {
+        throw std::invalid_argument{fmt::format(
+            "Negative edge weight [{}] between vertices [{}] -> [{}].",
+            edge_weight, current.id, neighbor)};
+      }
```

Comment:
[sidenote]: I know it is sub-optimal, but I would like to restrict the usage of `fmt` to just the `test` and `example` source code. That way we can keep the library independent of other libraries, such that clients can install it header-only.

Last time I checked GCC and Clang still had some issues with the `std::format` implementation, so until then we will have to resort to manual string concatenation 😅 

```suggestion
      if (edge_weight < 0) {
        throw std::invalid_argument{
            "Negative edge weight [" + edge_weight + "] between vertices [" + current.id + "] -> [" + neighbor + "]."};
      }
```

There might be some other places left in the codebase, so I can also go over this in a follow up.